app/src/split/bluetooth/peripheral.c

```diff
@@ -43,15 +43,55 @@ static const struct bt_data zmk_ble_ad[] = {
 
 static bool is_connected = false;
 
-static int start_advertising() {
-    return bt_le_adv_start(BT_LE_ADV_CONN, zmk_ble_ad, ARRAY_SIZE(zmk_ble_ad), NULL, 0);
+static void each_bond(const struct bt_bond_info *info, void *user_data) {
+    bt_addr_le_t *addr = (bt_addr_le_t *)user_data;
+
+    char addr_str[BT_ADDR_LE_STR_LEN];
+
+    bt_addr_le_to_str(&info->addr, addr_str, sizeof(addr_str));
+    LOG_DBG("EACH BOND: %s", addr_str);
+
+    if (bt_addr_le_cmp(&info->addr, BT_ADDR_LE_NONE) != 0) {
+        LOG_DBG("Capturing it!");
+        bt_addr_le_copy(addr, &info->addr);
+    }
+}
+
+static int start_advertising(bool low_duty) {
+    bt_addr_le_t central_addr = bt_addr_le_none;
+
+    bt_foreach_bond(BT_ID_DEFAULT, each_bond, &central_addr);
+
+    if (bt_addr_le_cmp(&central_addr, BT_ADDR_LE_NONE) != 0) {
+
+        char addr_str[BT_ADDR_LE_STR_LEN];
+
+        bt_addr_le_to_str(&central_addr, addr_str, sizeof(addr_str));
+        LOG_WRN("DIRECT to %s! low? %s", addr_str, (low_duty ? "yes" : "no"));
+        struct bt_le_adv_param adv_param = low_duty ? *BT_LE_ADV_CONN_DIR_LOW_DUTY(&central_addr)
+                                                    : *BT_LE_ADV_CONN_DIR(&central_addr);
+        return bt_le_adv_start(&adv_param, NULL, 0, NULL, 0);
+    } else {
+        return bt_le_adv_start(BT_LE_ADV_CONN, zmk_ble_ad, ARRAY_SIZE(zmk_ble_ad), NULL, 0);
+    }
 };
 
+static bool low_duty_advertising = false;
+
+static void advertising_cb(struct k_work *work) { start_advertising(low_duty_advertising); }
```

Comment:
Having a return value on `start_advertising` doesn't seem useful if we're just discarding it.

Reply:
Good call. Added actual checking w/ an error log message if we fail to advertise.